.travis.yml

```diff
@@ -6,9 +6,19 @@ env:
+  - secure: "kP5peDSNSOBxX1iIb1JsU4sJqf/1RVerEZl8/tGiJH5htYShT+qOgPI+GWoTOtx++nz7I05+VvLbloizN2uKbCc4/gH5HqsqBKLsLC5OLiSEaAq8Z3XJvcng1Gq2lWaM2xR1qk/EGwZJTx1QarRI1t97jkiFI3t9sReMUOrsHjAG77pgKFWTgp4u4kbEdgO7Eq3xzN2h+5T3mhypfX76sPXsg2tigVxPyODIQFf8VfGRwAJEN39GF8uL9spyW/V52oEBl5qrE6zZEJmLwqcZmkyzEbHTM1dI1vyEpWoW/cdDyBR7nwdiu8y8XrqvqGCSXquGTh0B3Vx/iHaq5gOsaacNm5RI47BQpLCA37BFciE4ENvxubRq45eN4sXeaqr210ODblDxG6xq8R25nLEmJC8jeV0MXbyuU10iJL5id0ZQMlDf9+9AfNQwUUwG3+Z5qQi2UL84R2fKAG2/i6CxWcUi/9VnqblCGb+AFunV/RAsVVmry/QiF1jxzZLcOHvvkLrNyW8/0pDD5nWX/Kh5gcwRe8Dq2YRdAoq9S7iPPiUgL/f8uwZdPNptJczfITDyv7o6ooat3rP6R38NueMR6MaG7pjzKzNklGzSTsH2RoxTryKeE+EOeQ/8BqkhJx6kyAMd87u2tiB65gtcXj26/s34kVpNUGV11cggbYVxA34="
+  - secure: "fkIMpdB51cpOgehQo+AUthfJrxOIZWUu917F/7GpmjOxyPcVO16pROIlSiC6J1GY+o/5sD1r8bMP4LKCZ7rtBM4SUaGqEsDTwt8wPUvJy21JVxd5NCjkeJVWR3FNFzhzcg5dk/WTbZnAuOW88wWgay9NvAXFoDF/JJkkAj6Z1XrNZFcrDLfZDdDdRZUVzc02o1iJQTpnZAEG7c1Q3dEDmLRkg1ZmOZaMW+pIQuJBEGqK1tpzQSpxen5L7nzJCcb1cAMSOqTHrVEY2fCf/RS6fVedbXzJetdUbnGe0bWBpRghuIKaTozedsMlFmUiHaicNz0WAXj2nKUQTETiPWHXBaahbwjT8J1ElRBKU2QSvHQ/Gy6JqM7c0WtnqE4zXfDqplMzCtw3AcrqcUZBKO9qeT/XsNRNnHag3Q7S6OWDk4RfAmQpE6u0J9c74fRPQrr9YDLcgPqwRfsZs0aGtoIAHiqWXmeK6JV/m2U1+p6MzU/jFcq6E2UB0AP0s2FxIDf+Ch6sdSgPoqy5+jvjyFSHjxikaHjsqkFCCJuWfULzayxciOG8g72jvmU3mlwHCXUm3H6+p5JaDcONYQFQJPWeYaW8xvu1W/6sr8n/V/UcP/wXFhV0SAvtxkdrRanc3mt2fqj736Q72guSPklxYH9PFXZMeROMyejHhsGQKB+M/tc="
 
 script:
 - docker build --target test --tag todo-app:test .
 - docker run todo-app:test todo_app/tests/test_trello.py
 - docker run --env-file .env.test todo-app:test todo_app/tests/test_app.py
-- docker run --env TRELLO_BOARD_ID --env TRELLO_AUTH --env TRELLO_TOKEN todo-app:test todo_app/tests_e2e/test_e2e.py
\ No newline at end of file
+- docker run --env TRELLO_BOARD_ID --env TRELLO_AUTH --env TRELLO_TOKEN todo-app:test todo_app/tests_e2e/test_e2e.py
+- echo $DOCKER_PASSWORD | docker login --username $DOCKER_USER --password-stdin
+- docker build --target production --tag $DOCKER_USER/todo-app:latest .
+- docker push $DOCKER_USER/todo-app:latest
+- docker login --username=_ --password=$HEROKU_API_KEY registry.heroku.com
+- docker tag $DOCKER_USER/todo-app:latest registry.heroku.com/lachlanpearse-todo-app/web
+- docker push registry.heroku.com/lachlanpearse-todo-app/web
+- heroku container:release web --app lachlanpearse-todo-app
```

Comment:
Note that here you're pushing your production image on _every_ branch build - that means you're deploying before the code has been reviewed, or even necessarily finished. Instead we want to restrict this so that we can easily control what branches it runs on (usually just main), and manage that separately to the tests which we do want to run regularly. A nice way to do that is using the [deploy](https://docs.travis-ci.com/user/deployment/script/) and possibly the before_deploy lifecycle stages